src/nextjournal/clerk/builder.clj

```diff
@@ -132,16 +132,20 @@
   (->> (cond paths (if (sequential? paths)
                      paths
                      (throw (ex-info "`:paths` must be sequential" {:paths paths})))
-             paths-fn (if (qualified-symbol? paths-fn)
-                        (try
-                          (if-let [resolved-var  (requiring-resolve paths-fn)]
-                            (let [resolved-paths (cond-> @resolved-var
-                                                   (fn? @resolved-var) (apply []))]
-                              (when-not (sequential? resolved-paths)
-                                (throw (ex-info (str "#'" paths-fn " must be sequential.") {:paths-fn paths-fn :resolved-paths resolved-paths})))
-                              resolved-paths)
-                            (throw (ex-info (str "#'" paths-fn " cannot be resolved.") {:paths-fn paths-fn}))))
-                        (throw (ex-info "`:path-fn` must be a qualified symbol pointing at an existing var." {:paths-fn paths-fn}))))
+             paths-fn (let [ex-msg "`:path-fn` must be a qualified symbol pointing at an existing var."]
+                        (when-not (qualified-symbol? paths-fn)
+                          (throw (ex-info ex-msg {:paths-fn paths-fn})))
+                        (let [resolved-var  (try (requiring-resolve paths-fn)
```

Comment:
`requiring-resolve` returns nil if the namespace exists but the var doesn't.
The nullpointer will get caught by the catch in line 143, but the error message could be better in that case imho.
I think that should also result in `"`:path-fn` must be a qualified symbol pointing at an existing var."